src/main/java/frc/robot/FieldConstants.java

```diff
@@ -0,0 +1,97 @@
+package frc.robot;
+
+
+import static edu.wpi.first.apriltag.AprilTagFields.k2024Crescendo;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.math.geometry.*;
+import edu.wpi.first.math.util.Units;
+import java.io.IOException;
+
+/**
+ * Contains various field dimensions and useful reference points. Dimensions are in meters, and sets
+ * of corners start in the lower left moving clockwise. <b>All units in Meters</b> <br>
+ * <br>
+ *
+ * <p>All translations and poses are stored with the origin at the rightmost point on the BLUE
+ * ALLIANCE wall.<br>
+ * <br>
+ * Length refers to the <i>x</i> direction (as described by wpilib) <br>
+ * Width refers to the <i>y</i> direction (as described by wpilib)
+ */
+public class FieldConstants {
+  public static double fieldLength = Units.inchesToMeters(651.223);
+  public static double fieldWidth = Units.inchesToMeters(323.277);
+  public static double wingX = Units.inchesToMeters(229.201);
+  public static double podiumX = Units.inchesToMeters(126.75);
+  public static double startingLineX = Units.inchesToMeters(74.111);
```

Comment:
why not `final`?

Reply:
additionally, should be IN SCREAMING_SNAKE_CASE

Reply:
I copied and pasted from somewhere else. I can add final modifiers too.